components/script/dom/bindings/xmlname.rs

```diff
@@ -10,15 +10,10 @@ use html5ever::{LocalName, Namespace, Prefix};
 
 /// Validate a qualified name. See https://dom.spec.whatwg.org/#validate for details.
 pub fn validate_qualified_name(qualified_name: &str) -> ErrorResult {
+    // Step 2.
     match xml_name_type(qualified_name) {
-        XMLName::InvalidXMLName => {
-            // Step 1.
-            Err(Error::InvalidCharacter)
-        },
-        XMLName::Name => {
-            // Step 2.
-            Err(Error::Namespace)
-        },
+        XMLName::InvalidXMLName => Err(Error::InvalidCharacter),
+        XMLName::Name => Err(Error::InvalidCharacter),
```

Comment:
My reading of https://dom.spec.whatwg.org/#validate says that XMLName shouldn't throw anything. How do the test results look if that's the case?

Reply:
Oh, I misread this as being "does not match Name", but that's not the case

Reply:
Good point, maybe there's more than just the exception-type to fix here. I'll find out.

Reply:
When I change it to let Name in instead of throwing InvalidCharacter for it, then some tests go from correctly throwing InvalidCharacter to incorrectly not throwing, like Document.implementation.createDocumentType("prefix::local", "", "") is expected to throw Invalid Character. "prefix::local" appears to match the Name production. The test seems more consistent with just "if qualifiedName does not match the QName production."

Reply:
However I should check the codepaths for those "" parts too, maybe those are what's supposed to cause the error this time.

Reply:
https://dom.spec.whatwg.org/#interface-documenttype nope those are totally allowed to be empty strings, the error has to be for the prefix::local part